.jenkins/build-sanity.groovy

```diff
@@ -35,28 +35,55 @@ node('cico-workspace') {
 			}
 			firstAttempt = false
 			cico = sh(
-				script: "cico node get -f value -c hostname -c comment --release=8 --retry-count=${cico_retries} --retry-interval=${cico_retry_interval}",
+				script: "cico node get -f value -c hostname -c comment --release=8 \
+							--retry-count=${cico_retries} --retry-interval=${cico_retry_interval}",
 				returnStdout: true
 			).trim().tokenize(' ')
 			env.CICO_NODE = "${cico[0]}.ci.centos.org"
 			env.CICO_SSID = "${cico[1]}"
+			env.CICO_NODE_SSH = "ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@${CICO_NODE}"
 		}
 	}
 
 	try {
 		stage('prepare bare-metal machine') {
 			sh 'scp -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no ./.jenkins/prepare.sh root@${CICO_NODE}:'
-			sh "ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@${CICO_NODE} ./prepare.sh --workdir=/opt/build/noobaa-core --gitrepo=${ci_git_repo} --ref=${ci_git_ref}"
+			sh "${CICO_NODE_SSH} ./prepare.sh --workdir=/opt/build/noobaa-core --gitrepo=${ci_git_repo} --ref=${ci_git_ref}"
+		}
+
+		stage('stop jobs from the same PR') {
+			def JobName = "noobaa-core_build-sanity"
+			jobs = sh (
+				script: "${CICO_NODE_SSH} 'cd /opt/build/noobaa-core/.jenkins/ && ./get_job_numbers.sh --jobName ${JobName} \
+							--currentBuild ${currentBuild.number} --JENKINS_URL ${JENKINS_URL}'",
+				returnStdout: true
+			).trim().tokenize(' ')
+			if ( jobs.isEmpty() == true ) {
+				println "Nothing to abort"
+			} else { 
+				for (JobNumber in jobs) {
+					int JobNumberint = JobNumber as int
+					println "Aborting ${JobName} ${JobNumber}"
```

Comment:
In addition to aborting, you need to call `cico node done <session>` to return the bare-metal system that was reserved. That probably means setting some additional metadata (the `CICO_SSID`) in the job, fetching it and running the command.

If a reserved machine is not returned, it will stay counted against the quota of the project, until the machine is force-returned by the cico scheduler (after 24h since reserving). That means, you may not be able to reserve machines if jobs get aborted regularly.

Reply:
after the abort, it goes into:
```
finally {
		stage('return bare-metal machine') {
			sh 'cico node done ${CICO_SSID}'
		}
	}
```

Reply:
oh, if that is the case, it's all fine!